NTumbleBit/PuzzleSolver/SolverServerSession.cs

```diff
@@ -346,11 +346,10 @@ private void AssertState(SolverServerStates state)
 			fulfill.Inputs.Add(new TxIn(offerCoin.Outpoint));
 			fulfill.Outputs.Add(new TxOut(offerCoin.Amount, cashout));
 
-			var fulfillScript = SolverScriptBuilder.CreateFulfillScript(NBitcoin.BuilderExtensions.BuilderExtension.DummySignature, solutions) + Op.GetPushOp(offerCoin.Redeem.ToBytes());
+			var fulfillScript = SolverScriptBuilder.CreateFulfillScript(NBitcoin.BuilderExtensions.BuilderExtension.DummySignature, solutions);
 			fulfill.Inputs[0].ScriptSig = fulfillScript + Op.GetPushOp(offerCoin.Redeem.ToBytes());
-
 		
-			fulfill.Outputs[0].Value -= fulfill.GetVirtualSize();
```

Comment:
Not sure about the previous changes, but the size should definitely not be deducted from the output value, so utActk to this.